### DISCUSSION TITLE
Questions about ROC curves

### DISCUSSION THREAD

**Discussion by StephenC07** [SBC-Utrecht/pytom-match-pick#245] [Questions]
Dear Marten, I wanted to reach out as I've been using PyTom on cryo-FIB milled samples (150nm-220nm thick lamellae) with crowded and packed densities. I'm a bit stuck interpreting the ROC curves. In these tomograms I see cylindrical shaped densities which look like proteasomes. When template matching the proteasome core (~750 kDa) using a 50Å low-pass template and tomogram filter, I see reasonable curves and RUCs (proteasome example attached, RUC=0.59), the Relion reconstruction looks like the proteasome core in C1 generated from a blob as the reference. When I plot the particles back onto the tomograms I see proteasome densities. Everything in the pipeline looks sensible.

![Globular_protein700kDa](https://github.com/user-attachments/assets/c69e1ea1-7195-4809-bf5b-9a0e46307e64)
![Proteasome](https://github.com/user-attachments/assets/e930d77e-e359-43ab-bc47-ac2e01ea9b68)

I attempted to find a similar sized, more globular looking protein (700 kDa) in the same tomograms. With this template I see matched high-contrast densities and high LCCmax values with slightly lower RUC scores. The plots tend to start with high FDR/low recall which I believe means there is little separation between the noise and extracted gaussian curves (globular ROC plot example attached). In a separate experiment as a control I used a template for a protein that SHOULD NOT be present (similar size) and found hits with similar statistics and even produced a rough noisy reconstruction of this template in C1 (same pipeline as the proteasome), but this surely must be a result of noise-aligning to the template.

The densities for the globular protein, Pytom found, look sensible (high-contrast), but it's hard to be sure as it's not as clear cut as it was for the cylinder shaped proteasome. I am worried the globular protein reconstruction is produced from noise in these crowded tomograms like the control, however the average is much cleaner and sharper and goes to 15Å resolution from two-half maps. So the average looks great, but I am sceptical. I know there are a few tricks I can try to validate this such as leaving out a part of our target structure from the template.

I wondered if you could help me with any guidance to better understand how we can more precisely interpret the ROC statistics to be more confident Pytom has picked at least some of the intended densities or not? I am curious about whether I can accept RUC values below 0.5? Would you also be able to help me better understand how the noise-curve in the bimodal plot is calculated? 

I wanted to also ask maybe a naive question about how the sigma value of the map is chosen for template matching in PyTom? One last thing, I also have another naive question as I noticed for some complexes I attempted to find with PyTom, I see much better statistics and reconstruction results when the template is heavily low-pass filtered at 60Å with the tomogram low-pass filtered at a lower 30Å. I know you have advised in the PyTom tutorial to keep low-pass filtering consistent and I wondered what your thoughts were on them being different?

I hope this message finds you well. Thank you for your help. :)

**Comment by McHaillet** [comment]
Hi Stephen, thanks a lot for posting the question here! I will respond in a few days, as I am a bit busy atm, but just letting you know I saw your post. Cheers!

**Comment by McHaillet** [comment]
Sorry for the late reply Stephen!

Overall, you diagnosed the confidence issues quite well. You can see for the proteasome case that some of the extracted points (blue) clearly fall above the background. That is, if we assume the default background cutoff, which only tolerates one FP from the background (roughly at 0.43). You might say that is strict, but that is the most certain statistical estimate that the software can give (still gold markers can show up there). 

In conclusion, it is absolutely possible that the reconstructions are biased, as the purpose of template matching is to filter for lookalikes. Probably the safest approach to assess bias is to leave out part of our target structure from the template (e.g., use only the large ribosome subunit for localization, Hrabe et al J. Struct. Biol. 2012, or remove an arbitrary part, Yu et al, J. Struct. Biol. 2024). In many applications, getting to a sub-tomogram reconstruction of much higher resolution than the input template is also reassuring. In your case the former would probably be most suitable: a subtomogram average of a ‘baited’ search should get you the missing segment back. 

Considering your question about the calculation of the noise curve: the mean is assumed to be zero and the standard deviation is fitted to the distribution together with the Gaussian representing the particles. Scipy curve fit optimizes this functions. By bounding the background distribution more strongly, i.e. fixing the mean at 0, the fit is 'ensured'. Hope that makes sense.
In contrast, for the curve calculated during the candidate extraction, the standard deviation is actually tracked during the template matching search. Still a mean of 0 is assumed, but the variance is tracked by summing all the scores at each iteration and dividing them by the full search space. They are summed because normally for variance you need to subtract the mean, but as it is already assumed to be 0, we can just directly sum. The number of true positives is so small compared to the number of background voxels, that we can relatively safely ignore those for this calculation.

**Comment by McHaillet** [comment]
> I wanted to also ask maybe a naive question about how the sigma value of the map is chosen for template matching in PyTom? One last thing, I also have another naive question as I noticed for some complexes I attempted to find with PyTom, I see much better statistics and reconstruction results when the template is heavily low-pass filtered at 60Å with the tomogram low-pass filtered at a lower 30Å. I know you have advised in the PyTom tutorial to keep low-pass filtering consistent and I wondered what your thoughts were on them being different?

Oh, and considering the final question. Hmm.. perhaps that is a bad assumption on my part. I guess if you already limited the signal in the template it might be redundant for the tomogram. Although I am surprised that the results are actually better if the tomogram is not filtered, I would expect them to be very similar. Is there anyway you could share that (via email is also fine)?
